tests/cpp/test_move_pad.cpp

```diff
@@ -0,0 +1,437 @@
+  auto s4 = IrBuilder::create<Val>(4.0);
+  auto s5 = IrBuilder::create<Val>(4.0);
+  auto s6 = sub(s4, s5);
+  TensorView* tv7 =
+      pad(tv3,
+          {IrBuilder::create<Val>(1L),
+           IrBuilder::create<Val>(0L),
+           IrBuilder::create<Val>(0L),
+           IrBuilder::create<Val>(0L)},
+          s6);
+
+  fusion->addInput(tv0);
+  fusion->addOutput(tv1);
+  fusion->addOutput(tv2);
+  fusion->addOutput(tv7);
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  at::Tensor t0 = at::randn({4, 10}, options);
+  std::vector<c10::IValue> aten_inputs = {t0};
+
+  FusionExecutorCache fec(std::move(fusion));
+  auto out_tensors = fec.runFusionWithInputs(aten_inputs);
+
+  testValidate(fec.fusion(), out_tensors, aten_inputs, __LINE__, __FILE__);
+}
+
+TEST_F(MovePadTest, NotMergeNegativePad) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  TensorView* tv0 = makeContigConcreteTensor({4, 10});
+  TensorView* tv1 =
+      pad(tv0,
+          {IrBuilder::create<Val>(0L),
+           IrBuilder::create<Val>(-2L),
+           IrBuilder::create<Val>(1L),
+           IrBuilder::create<Val>(1L)});
+  TensorView* tv2 =
+      pad(tv1,
+          {IrBuilder::create<Val>(0L),
+           IrBuilder::create<Val>(3L),
+           IrBuilder::create<Val>(4L),
+           IrBuilder::create<Val>(0L)});
+  TensorView* tv3 = relu(tv2);
+
+  fusion->addInput(tv0);
+  fusion->addOutput(tv3);
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  at::Tensor t0 = at::randn({4, 10}, options);
+  std::vector<c10::IValue> aten_inputs = {t0};
+
+  FusionExecutorCache fec(std::move(fusion));
+  auto out_tensors = fec.runFusionWithInputs(aten_inputs);
+
+  testValidate(fec.fusion(), out_tensors, aten_inputs, __LINE__, __FILE__);
+}
+
+TEST_F(MovePadTest, BooleanCat) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  TensorView* tv0 = makeContigConcreteTensor({4, 10}, DataType::Bool);
+  TensorView* tv1 = makeContigConcreteTensor({4, 10}, DataType::Bool);
+  TensorView* tv2 = makeContigConcreteTensor({2, 10}, DataType::Bool);
+  TensorView* tv3 = logical_and(tv0, tv1);
+  TensorView* tv4 = cat({tv3, tv2}, /*dim=*/0);
+
+  fusion->addInput(tv0);
+  fusion->addInput(tv1);
+  fusion->addInput(tv2);
+  fusion->addOutput(tv4);
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  at::Tensor t0 = at::randn({4, 10}, options) > 0.5;
+  at::Tensor t1 = at::randn({4, 10}, options) > 0.5;
+  at::Tensor t2 = at::randn({2, 10}, options) > 0.5;
+  std::vector<c10::IValue> aten_inputs = {t0, t1, t2};
+
+  FusionExecutorCache fec(std::move(fusion));
+  auto out_tensors = fec.runFusionWithInputs(aten_inputs);
+
+  FusionKernelRuntime* runtime = fec.getMostRecentKernelRuntime();
+  EXPECT_EQ(runtime->fusionSegments()->groups().size(), 1);
+
+  // ExpressionEvaluator is hitting an assert with dynamic value.
```

Comment:
File a bug and link it here?

Reply:
Priya was kind enough to support it later. I'll open an issue for her to look at.
:bow: